### ISSUE TITLE
[Play 2.4] Custom logback file will not load in DEV mode

### ISSUE THREAD

**Issue by rmmeans** [playframework/playframework#4946]
I'm trying to use a custom logback file in DEV mode so that my additional loggers that are not defined in the default logging file. One easy example of this is when shipping akka logs which I'm shipping to logback via the recommended method: 
``` 
akka {
  loggers = ["akka.event.slf4j.Slf4jLogger"]
  loglevel="DEBUG"
}
```
Per the documentation, I must add additional logger settings to my logback file to get this to work:
```xml
<!-- Set logging for all Akka library classes to INFO -->
<logger name="akka" level="INFO" />
<!-- Set a specific actor to DEBUG -->
<logger name="actors.MyActor" level="DEBUG" />
```

But my settings are not picked up DEV mode, I have to start the app up in prod mode to see my settings apply.

This may be related to issue #4856 - but my scenario is different as not even the logback.xml file by the default name in the conf directory is used in dev mode where it does get picked up when running the app in prod mode.

**Comment by rmmeans** [comment]
Ok - Figured out my problem. First off, the logger was not being configured when you use compile time DI. I have PR open to fix that: #4948. The documentation should also be updated for the Custom configuration to mention that in order to change the dev logger, just drop your own `conf/logback-play-dev.xml` file. It's important to note again however, that this only works when using compile time DI if #4948 is accepted or you initiate the `Logger.configure(environment)` yourself in your app loader.

**Comment by rmmeans** [comment]
Canceling this. See comment I made on the PR that I closed.
